### PR TITLE
clang-tidy fixes

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Signed-off-by: Alexey-N-Chernyshov <alexey.n.chernyshov@gmail.com>

### Description of the Change
A few fixes for clang-tidy.

Found a useful `gsl::narrow_cast<uint8_t>` that must be used instead of `static_cast` for lose-data conversions.

### CODE REVIEW

File: core/common/libp2p/cbor_stream.cpp [hunk 1]
```diff
@@ -24,7 +24,9 @@ namespace fc::common::libp2p {
                             WriteCallbackFunc cb) {
     stream_->write(*input,
                    input->size(),
-                   [input, cb{std::move(cb)}](auto written) { cb(written); });
+                   [input{std::move(input)}, cb{std::move(cb)}](auto written) {
```
Comment (turuslan) [inline core/common/libp2p/cbor_stream.cpp:27]: is there use-after-move in `*input` argument?

File: core/common/libp2p/cbor_stream.cpp [hunk 2]
```diff
@@ -40,8 +42,7 @@ namespace fc::common::libp2p {
             return cb(count.error());
           }
           self->buffer_.resize(self->size_ + count.value());
-          self->consume(gsl::make_span(self->buffer_).subspan(self->size_),
-                        std::move(cb));
+          self->consume(gsl::make_span(self->buffer_).subspan(self->size_), cb);
```
Comment (turuslan) [inline core/common/libp2p/cbor_stream.cpp:45]: why not move `cb`?

Comment (Alexey-N-Chernyshov) [reply core/common/libp2p/cbor_stream.cpp:45]: `std::move of the const variable 'cb' has no effect`.
`cb` is captured by value and is `const`.

File: core/common/libp2p/cbor_stream.cpp (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply core/common/libp2p/cbor_stream.cpp]: fixed

File: core/common/libp2p/cbor_stream.cpp (hunk 2 above)
Comment (turuslan) [reply core/common/libp2p/cbor_stream.cpp:43]: ok.  
can remove `move`.  
or add `mutable` to lambda.

**Review by turuslan — APPROVED** [review]

**Review by ortyomka — APPROVED** [review]

**Pull request merged**